driver-service/src/main/java/by/sergo/driverservice/service/impl/CarServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+        checkDate(dto.getYearOfManufacture());
+        return Optional.of(mapToEntity(dto))
+                .map(carRepository::save)
+                .map(this::mapToDto)
+                .orElseThrow(() -> new BadRequestException(ExceptionMessageUtil.getCanNotCreateCarMessage()));
+    }
+
+    @Override
+    @Transactional
+    public CarResponse update(Long id, CarCreateUpdateRequest dto) {
+        var existCar = getByIdOrElseThrow(id);
+        checkCarForUpdate(dto, existCar);
+        var carToSave = mapToEntity(dto);
+        carToSave.setId(id);
+        return Optional.of(carToSave)
+                .map(carRepository::save)
+                .map(this::mapToDto)
+                .orElseThrow(() -> new BadRequestException(ExceptionMessageUtil.getCanNotUpdateCarMessage()));
+    }
+
+    @Override
+    @Transactional
+    public CarResponse delete(Long id) {
+        var car = getByIdOrElseThrow(id);
+        carRepository.deleteById(id);
+        return mapToDto(car);
+    }
+
+    @Override
+    public CarResponse getById(Long id) {
+        var car = getByIdOrElseThrow(id);
+        return mapToDto(car);
+    }
+
+    @Override
+    public CarResponse getByDriverId(Long driverId){
+        var car = carRepository.getCarByDriverId(driverId);
+        var existCar = car
+                .orElseThrow(() -> new NotFoundException(ExceptionMessageUtil.getNotFoundMessage("Car", "driverId", driverId)));
+        return mapToDto(existCar);
+    }
+
+    @Override
+    public CarListResponse getAll(Integer page, Integer size, String orderBy) {
+        PageRequest pageRequest = getPageRequest(page, size, orderBy);
+        var responsePage = carRepository.findAll(pageRequest)
+                .map(this::mapToDto);
+        return CarListResponse.builder()
+                .cars(responsePage.getContent())
+                .page(responsePage.getPageable().getPageNumber() + 1)
+                .totalPages(responsePage.getTotalPages())
+                .size(responsePage.getContent().size())
+                .total((int) responsePage.getTotalElements())
+                .sortedByField(orderBy)
+                .build();
+    }
+
+    private PageRequest getPageRequest(Integer page, Integer size, String orderBy) {
+        if (page < 1 || size < 1) {
+            throw new BadRequestException(ExceptionMessageUtil.getInvalidRequestMessage(page, size));
+        } else if (orderBy != null) {
+            Arrays.stream(Car.class.getDeclaredFields())
+                    .map(Field::getName)
+                    .filter(s -> s.contains(orderBy.toLowerCase()))
+                    .findFirst()
+                    .orElseThrow(() -> new BadRequestException(ExceptionMessageUtil.getInvalidSortingParamRequestMessage(orderBy)));
+            return PageRequest.of(page - 1, size).withSort(Sort.by(Sort.Order.asc(orderBy.toLowerCase())));
+        } else {
+            return PageRequest.of(page - 1, size);
+        }
+    }
```

Comment:
Метод private PageRequest getPageRequest(Integer page, Integer size, String orderBy) можно вынести в отдельный утилитный класс, добавить в параметры Class clazz и использовать этот метод для разных сервисов не дублируя код
```suggestion
    private <T> PageRequest getPageRequest(Integer page, Integer size, String orderBy, Class<T> clazz) {
        if (page < 1 || size < 1) {
            throw new BadRequestException(
                    ExceptionMessageUtil.getInvalidRequestMessage(page, size));
        } else if (orderBy != null) {
            Arrays.stream(clazz.getDeclaredFields())
                    .map(Field::getName)
                    .filter(s -> s.contains(orderBy.toLowerCase()))
                    .findFirst()
                    .orElseThrow(() -> new BadRequestException(ExceptionMessageUtil.getInvalidSortingParamRequestMessage(orderBy)));
            return PageRequest.of(page - 1, size).withSort(Sort.by(Sort.Order.asc(orderBy.toLowerCase())));
        } else {
            return PageRequest.of(page - 1, size);
        }
    }
```